### PR TITLE
Evaluate `slice()` arguments in a loop

### PR DESCRIPTION
**Pull request by lionel-**
To provide more accurate error information

### CODE REVIEW

File: R/slice.R [hunk 1]
```diff
@@ -270,31 +270,42 @@ is_slice_call <- function(error_call) {
...
   }
 
   withCallingHandlers(
-    mask$eval_all(quo),
+    mask$eval_all(quo(impl(!!!dots))),
     error = function(cnd) {
-      bullets <- slice_bullets(cnd, error_call)
-      parent <- if (is_slice_call(error_call)) cnd else cnd$parent
+      if (index) {
+        local_error_context(dots = dots, .index = index, mask = mask)
```
Comment (romainfrancois) [inline R/slice.R:295]: the "context" is also used by `i = cnd_bullet_cur_group_label()`

File: R/slice.R [hunk 2]
```diff
@@ -270,31 +270,42 @@ is_slice_call <- function(error_call) {
...
-  UseMethod("slice_bullets")
-}
-
-#' @export
-slice_bullets.default <- function(cnd, error_call, ...) {
-  if (is_slice_call(error_call)) {
-    msg <- glue("Problem evaluating `... = {as_label(quo_squash(quo))}` . ")
+slice_bullets <- function(cnd, error_call, index) {
+  if (index && is_slice_call(error_call)) {
+    msg <- cnd_bullet_header("evaluating")
```
Comment (romainfrancois) [inline R/slice.R:308]: So maybe here, the header can be "Problem when combining" or something if `index` is 0

Comment (lionel-) [reply R/slice.R:308]: Is it granted that it's always a combining problem when `index` is 0?

File: R/slice.R (hunk 1 above)
Comment (lionel-) [reply R/slice.R:295]: hmm I don't think so, it seems to use a separate context.

**Comment by lionel-** [comment]
@romainfrancois Now includes changes discussed on Slack.

File: R/slice.R (hunk 2 above)
Comment (lionel-) [reply R/slice.R]: Now says "Problem while computing indices." when index is 0.

**Review by romainfrancois — APPROVED** [review]
I can follow up by wrapping the errors from `slice_combine()` so that they look similar

**Comment by lionel-** [comment]
> I can follow up by wrapping the errors from slice_combine() so that they look similar

Sounds good, thanks!

**Pull request merged**